src/main/java/net/dv8tion/jda/core/audio/AudioConnection.java

```diff
@@ -258,28 +266,33 @@ else if (sendHandler == null && sendSystem != null)
 
     private synchronized void setupReceiveSystem()
     {
```

Comment:
With your changes, if there is a "consumer" but the socket is either null or closed, nothing happens.
In the old implementation, if the socket was null/closed (no matter if "consumer" is set, there was cleanup (your else branch)